blocks/library/embed/index.js

```diff
@@ -19,17 +19,21 @@ registerBlock( 'core/embed', {
 		caption: children( 'figcaption' )
 	},
 
-	edit( { attributes, isSelected, setAttributes } ) {
+	edit( { attributes, setAttributes, focus, setFocus } ) {
 		const { url, title, caption } = attributes;
 
 		return (
 			<figure>
-				<iframe src={ url } title={ title } />
-				{ caption || isSelected ? (
+				<div className="iframe-overlay">
+					<iframe src={ url } title={ title } />
```

Comment:
I think we still could. In the above example, there's little difference between the `.embed-overlay` and the `:before` pseudo-element.

The problem with @jasmussen's mockup in https://github.com/WordPress/gutenberg/issues/483#issuecomment-297280468 is that it assumes we know that the embed is a video type. How we get this information is still a little unclear at this point. The embed block's implementation might be able to determine its own content type through proposed APIs like [the `oembed/1.0/proxy` endpoint](https://github.com/xwp/wordpress-develop/pull/225), which could return the provider and let us compare against a "known" mapping of provider -> type. This is not great either, though even server-side we don't really have a good sense of the _type_ of a supported embed provider.

Otherwise we could default to something generic.